app/modules/asset_groups/metadata.py

```diff
@@ -0,0 +1,320 @@
+            raise AssetGroupMetadataError(
+                'Incorrect num encounters in form submission, there must be exactly one'
+            )
+
+        from app.modules.users.models import User
+
+        if current_user is not None and not current_user.is_anonymous:
+            self.owner = current_user
+        else:
+            self.owner = User.get_public_user()
+
+        # individual fields in the message are all valid, now check that it's valid in total
+        self._validate_contents()
+        self.data_processed = CreateAssetGroupMetadata.DataProcessed.complete
+
+    def _validate_contents(self):
+        # Message was valid, is the user allowed to do so
+        from app.modules.users.models import User
+
+        if self.anonymous:
+            if self.bulk_upload:
+                raise AssetGroupMetadataError(
+                    'anonymous users not permitted to do bulk upload'
+                )
+            if self.owner_assignment:
+                raise AssetGroupMetadataError(
+                    'anonymous users not permitted to assign owners'
+                )
+
+            log.info(
+                f'Anonymous asset_group posted, submitter_email={self.submitter_email}'
+            )
+
+            # if not provided, submitter_guid is allowed to be null
+            if self.submitter_email is not None:
+                self.anonymous_submitter = User.find(email=self.submitter_email)
+
+                if (
+                    self.anonymous_submitter is not None
+                    and self.anonymous_submitter.is_active
+                ):
+                    # Active users must log in, no spoofing
+                    raise AssetGroupMetadataError(
+                        f'Anonymous submitter using active user email {self.submitter_email}; rejecting',
+                        'Invalid submitter data',
+                        403,
+                    )
+        elif self.bulk_upload:
+            for group in current_user.asset_groups:
+                if group.bulk_upload and not group.is_processed():
+                    # Only one unprocessed bulk upload allowed at a time
+                    raise AssetGroupMetadataError(
+                        f'Bulk Upload {group.guid} not processed, please finish this before creating new bulk upload'
+                    )
+
+        else:  # Form upload by logged in user
+            unprocessed_groups = 0
+            for group in current_user.asset_groups:
+                if not group.is_processed() and not group.bulk_upload:
+                    unprocessed_groups += 1
+            # TODO arbitrary limit chosen for now
+            if unprocessed_groups > 10:
+                raise AssetGroupMetadataError(
+                    f'You have {unprocessed_groups} Asset groups outstanding, please finish these first'
+                )
+
+    def _validate_sightings(self):
+        from app.extensions.tus import tus_upload_dir
+
+        sighting_num = 0
```

Comment:
Curious why not a len() on the `self.request['sightings']` here and likewise for the encounters instead of another couple variables, but probably no matter, just peas or carrots.

Reply:
The sighting and encounter _num fields are used to generate the error messages sent back to the user so that they know which encounter in which sighting is at fault. It's to aid the user to know which one needs to be fixed. 
I'll admit that it does feel a bit clunky but I couldn't think of a better way.